src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -5561,23 +5561,59 @@ void ReportSurfaceHeatBalance(EnergyPlusData &state)
     // This subroutine puts the reporting part of the HBSurface Module in one area.
 
     using SolarShading::ReportSurfaceShading;
-
+    auto &Surface(state.dataSurface->Surface);
     state.dataHeatBalSurf->SumSurfaceHeatEmission = 0.0;
-
+    for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+        if (Surface(SurfNum).ExtBoundCond == ExternalEnvironment) {
```

Comment:
If this loop is executed frequently, it may be worthwhile to make a std::vector<int> of all surfaces that have `ExternalEnvironment` as their boundary condition.

Reply:
The condition `Surface(SurfNum).ExtBoundCond == ExternalEnvironment` is used frequently but is always paired with and/or with other conditions. A lot of those happen in 'HTSurfaceFirst' to last or 'WinFirst` to last. 
As for this loop alone, it gets executed every zone time step. Not sure if it's frequent enough.

Reply:
Well ... do they occur frequently enough that we want to sort the Window surfaces within each Zone into Exterior and Interior Windows? And same for Opaque surfaces?

Reply:
Not sure if it's worth it. Half of the time the check is paired with `ExternalEnvironment || Ground` or  `ExternalEnvironment || OtherSideCoefNoCalcExt` or `ExternalEnvironment || SurfaceClass == wall`, and half of the time the condition is checked in an enclosure loop where we don't have the sorted zone surface priviledge.